Prox/Prox/RemoteConfigKeys.swift

```diff
@@ -87,16 +87,34 @@ class RemoteConfigProperty<T> {
 
     var value: T {
         let remoteConfig = FIRRemoteConfig.remoteConfig()
+
+        // rcv is never nil even if there is no key.
+        // rcv.numberValue? never fails either (in light of an invalid number, it defaults to zero).
+        // So, we get the string value, and try parsing it ourselves.
         let rcv = remoteConfig[key]
-        if let value = convert(rcv) {
-            return value
+        if let string = rcv.stringValue {
+            if let value = convert(string) {
+                return value
+            } else if string != "" {
+                // i.e. no default exists, and no value is specified.
+                NSLog("RemoteConfigKeys: Existing value for \(key) is not a valid \(type(of: defaultValue))")
+            }
         }
-        if let rcv = remoteConfig.defaultValue(forKey: key, namespace: nil),
-            let value = convert(rcv) {
-            print("RemoteConfigKeys: Default value for \(key) from embedded plist")
+
+        // TODO: save the last successful value in UserDefaults.standard, then fall back to it here,
+        // instead of falling back to what's in the plist.
+
+        // Check to see if RemoteConfigDefaults.plist has a parseable default
```

Comment:
I've added more comments.

TBC: I really don't want to be using the plist. It doesn't help with type info, has terrible failure conditions.

We should *not* use the plist, though there are compelling cases (very long strings) when we could use them. If I started from scratch, I would not support the plist for defaults.